### ISSUE TITLE
How do I wipe out the commit message, but maintain sha1, etc...

### ISSUE THREAD

**Issue by DarthHater** [google/copybara#28]
This is what I've ended up with:

```
# Update these references to your orginzations repos
sourceUrl = "git@github.com:sonatype/nexus-repository-p2-internal.git"
destinationUrl = "git@github.com:sonatype/nexus-repository-p2.git"

core.workflow(
    name = "default",
    origin = git.origin(
        url = sourceUrl,
        ref = "master",
    ),
    destination = git.destination(
        url = destinationUrl,
        fetch = "master",
        push = "master",
    ),
    mode = "ITERATIVE",
    # Change path to the folder you want to publish publicly
    origin_files = glob(["**"], exclude = ["copy.bara.sky"]),
    destination_files = glob(["**"]),

    authoring = authoring.pass_thru("DarthHater <jhesse@sonatype.com>"),
    transformations = [
      metadata.squash_notes("Changes for Project Nexus Repository - P2:\n"),
    ],
)
```

I'd like to wipe out the commit message or make it generic. Origin sha1 is good to have. Is that possible with my setup? I'm using squash_notes out of ignorance, currently.

**Comment by denian** [comment]
`metadata.squash_notes` has flags for showing/hiding the ref and the author, but not for the description. Maybe that's a useful thing to add.

For now you might be able to achieve what you want by adding a `metadata.scrubber()` transformation after the `squash_notes` one. Something like (haven't tested the regex):

```
    transformations = [
      metadata.squash_notes("Changes for Project Nexus Repository - P2:\n"),
      # Replace "sha1 Commit description" by "sha1"
      metadata.scrubber('^([a-f0-9]{40}).*$', replacement = '$1'),
    ],
```

**Comment by DarthHater** [comment]
I think description would be good. If you point me to the right code I can see about adding it!

**Comment by denian** [comment]
Yes, that'd be great!

We would need to add a new param after `show_author`, called `show_description`. See:
https://github.com/google/copybara/blob/5df56d858a431500069bda9c8d99dfc51b284819/java/com/google/copybara/transform/metadata/MetadataModule.java#L77

and then use it accordingly in `MetadataSquashNotes` to decide if the message should be printed or not.

**Comment by DarthHater** [comment]
I think I've got this in #29

**Comment by DarthHater** [comment]
Closed in #32
